lib/pagseguro/collection_base.rb

```diff
@@ -0,0 +1,41 @@
+module PagSeguro
+  class CollectionBase
+    extend CollectionManagers::MacroMethods
+    extend Forwardable
+    include Enumerable
+
+    def_delegators :@store, :size, :clear, :empty?, :any?, :each
+
+    def initialize
+      @store = []
+    end
+
+    def collection_manager
+      CollectionManagers::GenericManager
+    end
+
+    def collection_type
+      eval singularize(self.class.to_s)
```

Comment:
So How can I call the collection type throught the class name, for example...
Whether I have a `PagSeguro::Documents` my collection type should be `PagSeguro::Document` are two different classes.